### PR TITLE
Add an API method for determining the best status

### PR DESCRIPTION
**Pull request by sethvargo**
Given a list of HealthChecks, this determines the "best" status for the collective group. This is useful for nodes and services, which may have multiple checks associated with them.

This is the initial groundwork for https://github.com/hashicorp/consul/issues/802

### CODE REVIEW

File: api/health.go [hunk 1]
```diff
@@ -25,6 +36,52 @@ type HealthCheck struct {
...
+
+		switch check.Status {
+		case HealthPassing:
+			passing = true
+		case HealthWarning:
+			warning = true
+		case HealthCritical:
+			critical = true
+		default:
+			log.Printf("[WARN] unknown status %q", check.Status)
```
Comment (slackpad) [inline api/health.go]: I'd drop this because I don't think we log in here usually.

File: api/health.go [hunk 2]
```diff
@@ -25,6 +36,52 @@ type HealthCheck struct {
...
+type HealthChecks []*HealthCheck
+
+// AggregatedStatus returns the "best" status for the list of health checks.
+// Because a given entry may have many service and node-level health checks
+// attached, this function determines the best representative of the status as
+// as single string using the following heuristic:
+//
+//  maintenance > critical > warning > passing
+//
+func (c HealthChecks) AggregatedStatus() string {
```
Comment (slackpad) [inline api/health.go]: I'd make this have an error return and bail with an error for the `return ""` case.

Comment (sethvargo) [inline api/health.go]: Talked about this and just returning the empty string 😄

**Comment by slackpad** [comment]
LGTM

**Pull request merged**